components/scheduler/statemachine.go

```diff
@@ -90,11 +82,14 @@ func (stateMachineDefinition) Deserialize(d []byte) (any, error) {
 	if err := proto.Unmarshal(d, state); err != nil {
 		return nil, serialization.NewDeserializationError(enumspb.ENCODING_TYPE_PROTO3, err)
 	}
-	return Scheduler{state}, nil
+	return &Scheduler{
```

Comment:
The ensureFields call below should be called here.
But you can probably just get rid of that method and move it here and on instantiation. The scheduler instance should always assume everything is initialized.

Reply:
Pretty sure all deserialized scheduler instances would have been previously serialized, which means that the empty fields would already be filled in during initialization. (ensureFields only manipulates proto state)

We are going to move the rest to executor anyways.